client_update.go

```diff
@@ -86,8 +86,10 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string) (io.ReadCloser,
 	}
 
 	if r.ContentLength < 0 {
+		defer r.Body.Close()
```

Comment:
Crap. You are right. I did not follow the logic. In this case I think you can skip `defer` and close the stream explicitly. What is more, you are right that we need to close the stream in `func (u *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, bool)`.
Can you add something like below:

```
func (u *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, bool) {

	defer u.imagein.Close()

	// start deployment logging
	DeploymentLogger.Enable(u.update.ID)

```